ldm/dream/pngwriter.py

```diff
@@ -32,54 +33,31 @@ def unique_prefix(self):
 
     # saves image named _image_ to outdir/name, writing metadata from prompt
     # returns full path of output
-    def save_image_and_prompt_to_png(self, image, prompt, name):
+    def save_image_and_prompt_to_png(self, image, prompt, metadata, name):
         path = os.path.join(self.outdir, name)
         info = PngImagePlugin.PngInfo()
         info.add_text('Dream', prompt)
+        info.add_text('sd-metadata', json.dumps(metadata))
```

Comment:
So, the 'Dream' tag is not duplicated into the 'sd-metadata' tag?

Reply:
I believe the RFC is meant to be SD-platform independent. So I suppse our dream REPL command need not be included. Or, maybe it should be included in one of the `extra` areas? That sounds better...

Reply:
:headdesk: I read "it does not write all the required top-level metadata" in the comments and it went completely over my head, however, I see that lots of stuff has been added.